File: app/src/main/java/com/example/weizhenbin/wangebug/views/LinearRecyclerView.java
package com.example.weizhenbin.wangebug.views;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.annotation.Nullable;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;

import com.example.weizhenbin.wangebug.R;

/**
 * Created by weizhenbin on 2018/11/1.
 */
public class LinearRecyclerView extends RecyclerView {
    public LinearRecyclerView(Context context) {
        this(context,null);
    }

    public LinearRecyclerView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs,0);
    }

    public LinearRecyclerView(Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init(context);
    }

    private void init(Context context) {
        setLayoutManager(new LinearLayoutManager(context,LinearLayoutManager.VERTICAL,false));
        DividerItemDecoration itemDecoration=new DividerItemDecoration(context,DividerItemDecoration.VERTICAL);
        Drawable drawable=getResources().getDrawable(R.drawable.divider_line);
        itemDecoration.setDrawable(drawable);
        addItemDecoration(itemDecoration);
    }
}

File: app/src/main/java/com/example/weizhenbin/wangebug/views/LinearRecyclerView.kt
package com.example.weizhenbin.wangebug.views

import android.content.Context
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.AttributeSet
import com.example.weizhenbin.wangebug.R

/**
 * Created by weizhenbin on 2018/11/1.
 */
class LinearRecyclerView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) : RecyclerView(context, attrs, defStyle) {

    init {
        layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        val itemDecoration = DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
        val drawable = resources.getDrawable(R.drawable.divider_line)
        itemDecoration.setDrawable(drawable)
        addItemDecoration(itemDecoration)
    }

}
